### ISSUE TITLE
Conditional Nested Effects

### ISSUE THREAD

**Issue by peerreynders** [WebReflection/usignal#3]
Not sure if this is relevant for µsignal, please close if not.

@preact/signals output:

```shell
Activating: red
Red is off
Red is on
Activating: green
Green is on
Green is off
Activating: blue
Blue is off
Blue is on
Activating: red
Red is on
```

µsignal

```shell
Activating: red
Red is off
Red is on
Activating: green
Activating: blue
Activating: red
```

For details see:
https://gist.github.com/peerreynders/a942a2b4dac89f2e5eda456693a720ea

**Comment by WebReflection** [comment]
not sure what I’m looking at but dispose is returned by effect in here (latest version)

**Comment by WebReflection** [comment]
also effect is async … for the conditional nested in general I’m working on it but the assumption is that conditional can be the amount of effects but they are all same “brand” as in same kind of component (loop or something).

if this assumption doesn’t hold real world use cases then I need to re-think nested loops

**Comment by WebReflection** [comment]
P.S. this is relevant and a valid issue, thanks for filing it

**Comment by WebReflection** [comment]
@peerreynders fixed in 0.4.6

FWIWI I was already working on the *way too greedy effects cache* system but the truth is that there's no way around so ... I've added all comments and your test to the test folder which I believe now is reporting the correct things.

This might have a slightly impact on performance due (mostly) always re-computed effects *but* correctness has definitively the top priority over code size or perf.

Let me know if something is off now, thanks.

**Comment by WebReflection** [comment]
@peerreynders btw apparently [preact doesn't support nested effects](https://github.com/preactjs/signals/issues/152#issuecomment-1247693864) so it looks like it was working by accident.
